sqlglot/dialects/snowflake.py

```diff
@@ -746,6 +746,16 @@ class Parser(parser.Parser):
             "TO_JSON": exp.JSONFormat.from_arg_list,
             "VECTOR_L2_DISTANCE": exp.EuclideanDistance.from_arg_list,
             "ZEROIFNULL": _build_if_from_zeroifnull,
+            "LIKE": lambda args: exp.Escape(
+                this=exp.Like(this=args[0], expression=args[1]), expression=seq_get(args, 2)
+            )
+            if seq_get(args, 2)
+            else exp.Like(this=args[0], expression=args[1]),
+            "ILIKE": lambda args: exp.Escape(
+                this=exp.ILike(this=args[0], expression=args[1]), expression=seq_get(args, 2)
+            )
+            if seq_get(args, 2)
+            else exp.ILike(this=args[0], expression=args[1]),
```

Comment:
Let's factor this logic out into a helper:

```python
def _build_like(expr_type: t.Type[E]) -> t.Callable[[t.List], E]:
    def _builder(args: t.List) -> E:
        ...
    return _builder
```

It should be able to express this logic in a cleaner fashion, and then repeat it for both `"LIKE"` and `"ILIKE"`, similar to other functions that we map to similar callables returning closure builders.